### PR TITLE
Feature/ios background location

### PR DESCRIPTION
**Pull request by Wackymax**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)

Adds an additional option to Apple settings that allows the user to configure the background location indicator that needs to be set to true if the application is started up in the background by some other process and it then requests a location.

### :arrow_heading_down: What is the current behavior?

Cannot be set currently.

### :new: What is the new behavior (if this is a feature change)?

Exposing the CLocation manager setting to the user

### :boom: Does this PR introduce a breaking change?

No it doesn't.

### :bug: Recommendations for testing

One will have to start the application in the background through a system event or reminder and then request the location.

### :memo: Links to relevant issues/docs

N/A

### :thinking: Checklist before submitting

- [X] I made sure all projects build.
- [X] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy](https://dart.dev/tools/pub/versioning).
- [X] I updated CHANGELOG.md to add a description of the change.
- [X] I followed the style guide lines ([code style guide](https://github.com/Baseflow/flutter-geolocator/blob/master/CONTRIBUTING.md)).
- [X] I updated the relevant documentation.
- [X] I rebased onto current `master`.

### CODE REVIEW

File: geolocator_apple/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 2.1.1
+
+- Added additional option to Apple settings to allow the user to configure the background location indicator of CLocationManager
```
Comment (mvanbeusekom) [inline geolocator_apple/CHANGELOG.md:3]: Nit: sentence should end with period.
```suggestion
- Added additional option to Apple settings to allow the user to configure the background location indicator of CLocationManager.
```

File: geolocator_apple/apple/Classes/Handlers/GeolocationHandler.h [hunk 2]
```diff
@@ -24,7 +24,8 @@ typedef void (^GeolocatorResult)(CLLocation *_Nullable location);
 
 - (void)startListeningWithDesiredAccuracy:(CLLocationAccuracy)desiredAccuracy
                            distanceFilter:(CLLocationDistance)distanceFilter
-        pauseLocationUpdatesAutomatically:(BOOL)pauseLocationUpdatesAutomatically
+                            pauseLocationUpdatesAutomatically:(BOOL)pauseLocationUpdatesAutomatically
+                            showBackgroundLocationIndicator:(BOOL)showBackgroundLocationIndicator
```
Comment (mvanbeusekom) [inline geolocator_apple/apple/Classes/Handlers/GeolocationHandler.h:28]: In Objective-C it is common to align parameters on the colon `:`, please align the parameters here.
```suggestion
        pauseLocationUpdatesAutomatically:(BOOL)pauseLocationUpdatesAutomatically
          showBackgroundLocationIndicator:(BOOL)showBackgroundLocationIndicator
```

File: geolocator_apple/apple/Classes/Handlers/GeolocationHandler.m [hunk 3]
```diff
@@ -69,12 +69,20 @@ - (void)requestPositionWithDesiredAccuracy:(CLLocationAccuracy)desiredAccuracy
 - (void)startListeningWithDesiredAccuracy:(CLLocationAccuracy)desiredAccuracy
                            distanceFilter:(CLLocationDistance)distanceFilter
         pauseLocationUpdatesAutomatically:(BOOL)pauseLocationUpdatesAutomatically
+                             showBackgroundLocationIndicator: (BOOL) showBackgroundLocationIndicator
```
Comment (mvanbeusekom) [inline geolocator_apple/apple/Classes/Handlers/GeolocationHandler.m:72]: Same here, please ensure parameters are aligned correctly.
```suggestion
          showBackgroundLocationIndicator: (BOOL) showBackgroundLocationIndicator
```

File: geolocator_apple/apple/Classes/Handlers/GeolocationHandler.m [hunk 4]
```diff
@@ -69,12 +69,20 @@ - (void)requestPositionWithDesiredAccuracy:(CLLocationAccuracy)desiredAccuracy
...
-  
+    
+    self.errorHandler = errorHandler;
+    self.resultHandler = resultHandler;
+
+#if TARGET_OS_IOS
+    if (@available(iOS 11.0, macOS 11.0, *)) {
+        self.locationManager.showsBackgroundLocationIndicator = showBackgroundLocationIndicator;
+    }
+#endif
```
Comment (mvanbeusekom) [inline geolocator_apple/apple/Classes/Handlers/GeolocationHandler.m:84]: Please also align the code block here, we use two space per indentation.
```suggestion
  self.errorHandler = errorHandler;
  self.resultHandler = resultHandler;

#if TARGET_OS_IOS
  if (@available(iOS 11.0, macOS 11.0, *)) {
      self.locationManager.showsBackgroundLocationIndicator = showBackgroundLocationIndicator;
  }
#endif
```

File: geolocator_apple/apple/Classes/Handlers/PositionStreamHandler.m [hunk 5]
```diff
@@ -49,10 +49,13 @@ - (FlutterError *_Nullable)onListenWithArguments:(id _Nullable)arguments eventSi
   CLLocationDistance distanceFilter = [LocationDistanceMapper toCLLocationDistance:(NSNumber *)arguments[@"distanceFilter"]];
   NSNumber* pauseLocationUpdatesAutomatically = arguments[@"pauseLocationUpdatesAutomatically"];
   CLActivityType activityType = [ActivityTypeMapper toCLActivityType:(NSNumber *)arguments[@"activityType"]];
+
+  NSNumber* showBackgroundLocationIndicator = arguments[@"showBackgroundLocationIndicator"];
 	      
   [[weakSelf geolocationHandler] startListeningWithDesiredAccuracy:accuracy
                                                     distanceFilter:distanceFilter
                                  pauseLocationUpdatesAutomatically:pauseLocationUpdatesAutomatically && [pauseLocationUpdatesAutomatically boolValue]
+                                 showBackgroundLocationIndicator:showBackgroundLocationIndicator && [showBackgroundLocationIndicator boolValue]
```
Comment (mvanbeusekom) [inline geolocator_apple/apple/Classes/Handlers/PositionStreamHandler.m:58]: Same here:

```suggestion
                                   showBackgroundLocationIndicator:showBackgroundLocationIndicator && [showBackgroundLocationIndicator boolValue]
```

File: geolocator_apple/example/ios/RunnerTests/GeolocationHandlerTests.m [hunk 6]
```diff
@@ -136,7 +136,8 @@ - (void)testStartListeningShouldNotStopListeningWhenListeningToStream {
   XCTestExpectation *expectation = [self expectationWithDescription:@"expect result return third location"];
   [_geolocationHandler startListeningWithDesiredAccuracy: kCLLocationAccuracyBest
                                           distanceFilter:0
-                       pauseLocationUpdatesAutomatically:NO
+                                        pauseLocationUpdatesAutomatically:NO
+                                        showBackgroundLocationIndicator:NO
```
Comment (mvanbeusekom) [inline geolocator_apple/example/ios/RunnerTests/GeolocationHandlerTests.m:140]: Same:
```suggestion
                       pauseLocationUpdatesAutomatically:NO
                         showBackgroundLocationIndicator:NO
```

File: geolocator_apple/example/ios/RunnerTests/GeolocationHandlerTests.m [hunk 7]
```diff
@@ -181,6 +183,7 @@ - (void)testStartListeningShouldNotReportErrorOnErrorDomainAndErrorLocationUnkno
   [_geolocationHandler startListeningWithDesiredAccuracy: kCLLocationAccuracyBest
                                           distanceFilter:0
                        pauseLocationUpdatesAutomatically:NO
+                       showBackgroundLocationIndicator:NO
```
Comment (mvanbeusekom) [inline geolocator_apple/example/ios/RunnerTests/GeolocationHandlerTests.m:186]: Same:
```suggestion
                         showBackgroundLocationIndicator:NO
```

File: geolocator_apple/pubspec.yaml [hunk 8]
```diff
@@ -20,7 +20,7 @@ dependencies:
   flutter:
     sdk: flutter
 
-  geolocator_platform_interface: ^4.0.0
+  geolocator_platform_interface: ^4.0.0+1
```
Comment (mvanbeusekom) [inline geolocator_apple/pubspec.yaml:23]: This should be version 4.0.3:
```suggestion
  geolocator_platform_interface: ^4.0.3
```

**Review by mvanbeusekom — CHANGES_REQUESTED** [review]
The code looks good, just some formatting changes and the correct version number of the `geolocator_platform_interface` package.

**Comment by Wackymax** [comment]
Applied as per suggestions.

**Pull request merged**